### PR TITLE
Doc fixes

### PR DESCRIPTION
**Pull request by hakonfam**
I'm not satisfied with how I explain the secondary magic being unset, please come with your ideas :)

### CODE REVIEW

**Comment by nvlsianpu** [comment]
image swaps state enumeration looks good.

File: docs/design.md [hunk 1]
```diff
@@ -606,7 +606,7 @@ process is presented below.
 The boot loader swaps the contents of the two image slots for two reasons:
 
   * User has issued a "set pending" operation; the image in the secondary slot
-    should be run once (state II) or repeatedly (state III), depending on
+    should be run once (state I) or repeatedly (state II), depending on
     whether a permanent swap was specified.
   * Test image rebooted without being confirmed; the boot loader should
     revert to the original image currently in the secondary slot (state IV).
```
Comment (davidvincze) [inline docs/design.md]: Would you mind updating the rest of the state denotations? 
```suggestion
    revert to the original image currently in the secondary slot (state III).
```

**Comment by hakonfam** [comment]
> > I'm not satisfied with how I explain the secondary magic being unset, please come with your ideas :)
> 
> What is the reason - I don't catch?

I couldn't find a better place to put it, but I would like to have it written somewhere in the general description of the update procedure.

The way it is currently would suffice for the docs to be consistent though, so I guess it is OK :)

**Comment by nvlsianpu** [comment]
Should the `unset` concept be explained?

**Comment by nvlsianpu** [comment]
@utzig could you have a look?

**Comment by utzig** [comment]
> @utzig could you have a look?

It's on my list :rofl:

File: docs/design.md [hunk 2]
```diff
@@ -675,12 +675,14 @@ happened when a swap was requested:
...
             primary_slot.magic = BOOT_MAGIC
+            secondary_slot.magic = UNSET
             primary_slot.image_ok = Unset)
 
     * permanent:
         o Write primary_slot.copy_done = 1
         (swap caused the following values to be written:
             primary_slot.magic = BOOT_MAGIC
+            secondary_slot.magic = UNSET
             primary_slot.image_ok = 0x01)
```
Comment (utzig) [inline docs/design.md]: Just after beginning a swap the secondary slot's trailer is erased so `image_ok` is erased as well (although there's not particular need to mention it here).

**Pull request merged**